lib/komponent/core/component_path_resolver.rb

```diff
@@ -0,0 +1,66 @@
+module Komponent
+  class ComponentPathResolver
+    attr_accessor :path
+
+    def initialize(component)
+      @component = component
+      resolve
+    end
+
+    protected
+
+    def resolve
+      component_found = nil
+
+      component_paths.each do |component_path|
+        component_found = find_component(component_path)
+        break if component_found
+      end
```

Comment:
It would be more readable if we had something like `root_path = component_paths.find { |path| path_has_component?(path, @component)`